### ISSUE TITLE
Create SQL API for delta restore command

### ISSUE THREAD

**Issue by Maks-D** [delta-io/delta#891]
It will be nice to have SQL API for delta restore command (similar to Databricks delta (https://docs.databricks.com/spark/latest/spark-sql/language-manual/delta-restore.html)

Will be unblocked after merging #863

**Comment by rahulsmahadev** [comment]
Hi @Maks-D I am not sure if spark has that keyword in it's grammar ? It may not be possible to support SQL syntax here. For the same reason we do not support `VERSION AS OF` in Delta Lake with Spark.

**Comment by Maks-D** [comment]
I am going to start working on this task
